File: packages/api/app/schemas/task.py
from __future__ import annotations
from pydantic import BaseModel, Field
from datetime import datetime, date as date_type
from typing import Optional


class TaskCreate(BaseModel):
    title: str
    description: Optional[str] = None
    assignee: Optional[str] = None
    due_date: Optional[date_type] = None
    priority: str = "medium"
    source_meeting_id: Optional[str] = None
    key_result_id: Optional[str] = None
    source_segment_start: Optional[int] = None
    source_segment_end: Optional[int] = None


class TaskUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    assignee: Optional[str] = None
    due_date: Optional[date_type] = None
    priority: Optional[str] = None
    status: Optional[str] = None
    key_result_id: Optional[str] = None
    source_segment_start: Optional[int] = None
    source_segment_end: Optional[int] = None


class TaskComplete(BaseModel):
    result_description: Optional[str] = None
    result_links: Optional[list[str]] = None
    attachments: Optional[list[str]] = None
    impact_conclusion: Optional[str] = None


class SourceMeeting(BaseModel):
    meeting_id: str
    title: str
    date: Optional[date_type] = None


class TaskResponse(BaseModel):
    task_id: str = Field(alias="id")
    title: str
    description: Optional[str] = None
    assignee: Optional[str] = None
    due_date: Optional[date_type] = None
    priority: str
    status: str
    source_meeting: Optional[SourceMeeting] = None
    key_result_id: Optional[str] = None
    okr: Optional[dict] = None
    source_segment_start: Optional[int] = None
    source_segment_end: Optional[int] = None
    created_at: datetime
    updated_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None

    class Config:
        from_attributes = True
        populate_by_name = True


class TaskStatistics(BaseModel):
    total_tasks: int
    todo_count: int
    in_progress_count: int
    done_count: int
    overdue_count: int


class TaskBoardResponse(BaseModel):
    tasks: list[TaskResponse]
    statistics: TaskStatistics

File: packages/shared/src/types/task.ts
export interface SourceMeeting {
  meeting_id: string
  title: string
  date?: string
}

export interface Task {
  id: string
  title: string
  description?: string
  assignee?: string
  due_date?: string
  priority: 'high' | 'medium' | 'low'
  status: 'todo' | 'in_progress' | 'done' | 'blocked'
  source_meeting?: SourceMeeting
  key_result_id?: string | null
  okr?: {
    project_id: string
    project_name: string
    objective_id: string
    objective_title: string
    key_result_id: string
    key_result_title: string
    progress: number
  }
  source_segment_start?: number
  source_segment_end?: number
  created_at: string
  updated_at?: string
  completed_at?: string
}

export interface TaskStatistics {
  total_tasks: number
  todo_count: number
  in_progress_count: number
  done_count: number
  overdue_count: number
}

export interface TaskBoardResponse {
  tasks: Task[]
  statistics: TaskStatistics
}

export interface TaskCreate {
  title: string
  description?: string
  assignee?: string
  due_date?: string
  priority?: 'high' | 'medium' | 'low'
  source_meeting_id?: string
  key_result_id?: string | null
  source_segment_start?: number
  source_segment_end?: number
}

export interface TaskUpdate {
  title?: string
  description?: string
  assignee?: string
  due_date?: string
  priority?: 'high' | 'medium' | 'low'
  status?: 'todo' | 'in_progress' | 'done' | 'blocked'
  key_result_id?: string | null
  source_segment_start?: number
  source_segment_end?: number
}
